src/web/theme/pages/OrderReference/EnhanceOrderReference.js

```diff
@@ -0,0 +1,49 @@
+import { graphql } from "react-apollo";
+import compose from "recompose/compose";
+import withState from "recompose/withState";
+import withHandlers from "recompose/withHandlers";
+
+const orders = [
+  { id: 468513544, products: [{ sku: "epi_guitar" }] },
+  { id: 123456789, products: [{ sku: "epi_guitar" }, { sku: "epi_guitar02" }] },
+  {
+    id: 987654321,
+    products: [
+      { sku: "epi_guitar" },
+      { sku: "epi_guitar02" },
+      { sku: "epi_guitar" },
+      { sku: "epi_guitar02" }
+    ]
+  },
+  {
+    id: 654987321,
+    products: [
+      { sku: "epi_guitar" },
+      { sku: "epi_guitar" },
+      { sku: "epi_guitar" },
+      { sku: "epi_guitar" }
+    ]
+  },
+  {
+    id: 654321987,
+    products: [
+      { sku: "epi_guitar" },
+      { sku: "epi_guitar" },
+      { sku: "epi_guitar" }
+    ]
+  }
+];
+
+export default () =>
+  compose(
+    withState("customerOrder", "setCustomerOrder", { id: 0 }),
+    withState("noProductFound", "setnoProductFound", false),
+    withHandlers({
+      checkCustomerID: props => e =>
+        orders.filter(order => order.id == e.target.value).length > 0
```

Comment:
It would be better to have an actual function body here (instead of an implicit return) so that you don't have to repeat your `filter`. Also you are only interested by the first match, so `find` is more appropriate